libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt

```diff
@@ -205,11 +205,35 @@ class BasicAssertionsTest {
     fun testExpectFails() {
         checkFailedAssertion { expect(1) { 2 } }
     }
+
+    @Test
+    fun testAssertIs() {
+        val s: Any = "test"
+        assertIs<String>(s)
+        assertEquals(4, s.length)
+    }
+
+    @Test
+    fun testAssertIsFails() {
+        val message = "Expected value to be of type <kotlin.Int>, actual <class kotlin.String>."
+        checkFailedAssertion(message) { assertIs<Int>("test") }
+    }
+
+    @Test
+    fun testAssertNotIs() {
+        assertNotIs<Int>("test")
+    }
+
+    @Test
+    fun testAssertNotIsFails() {
+        val message = "Expected value to not be of type <kotlin.Int>."
+        checkFailedAssertion(message) { assertNotIs<Int>(1) }
+    }
 }
 
 
-internal fun checkFailedAssertion(assertion: () -> Unit): AssertionError {
-    return assertFailsWith<AssertionError> { withDefaultAsserter(assertion) }
+internal fun checkFailedAssertion(message: String? = null, assertion: () -> Unit): AssertionError {
+    return assertFailsWith(message)<AssertionError> { withDefaultAsserter(assertion) }
```

Comment:
This is a message for failed `assertFailsWith`, not the message to check in the successfully caught exception.
Use the `AssertionError` returned from `checkFailedAssertion` to verify its properties in particular tests.

Reply:
🤦‍♂️ thanks!